matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/GroupSyncHandler.kt

```diff
@@ -37,7 +37,7 @@ internal class GroupSyncHandler @Inject constructor() {
         data class LEFT(val data: Map<String, Any>) : HandlingStrategy()
     }
 
-    fun handle(
+    suspend fun handle(
```

Comment:
We cannot have suspendable here since we are caling this inside a Realm transaction which does not support thread change.

Inside maybe you can replace the `MutableLiveData` here https://github.com/vector-im/element-android/blob/main/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncRequestStateTracker.kt#L29 by a `Flow` instead?

Reply:
Sure, I'm curious - what about running the code inside the `handleResponse` function (https://github.com/vector-im/element-android/blob/3b2f9d9404129aafd477f1ad91f88a0c7d21a411/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt#L69)  on a single-thread executor (`Executors.newSingleThreadExecutor()`)? Would it not fix the thread switching related problems?

Reply:
Btw. I see that `Realm.WRITE_EXECUTOR` which is used inside the transaction is already single threaded...

Reply:
The `Flow`-refactoring is done